Brooks/LanguageRenderer/NewCSharpRenderer.hs

```diff
@@ -0,0 +1,561 @@
+    endStatement = return semi
+    endStatementLoop = return empty
+
+    include _ = return $ text "using"
+    inherit = return colon
+
+    list _ = return $ text "List"
+    listObj = return new
+
+    blockStart = return lbrace
+    blockEnd = return rbrace
+
+    ifBodyStart = blockStart
+    elseIf = return $ text "else if"
+    
+    iterForEachLabel = return $ text "foreach"
+    iterInLabel = return $ text "in"
+
+    commentStart = return doubleSlash
+    
+    printFunc = return $ text "Console.Write"
+    printLnFunc = return $ text "Console.WriteLine"
+    printFileFunc = fmap (printFileDocD "Write")
+    printFileLnFunc = fmap (printFileDocD "WriteLine")
+
+instance PermanenceSym CSharpCode where
+    type Permanence CSharpCode = Doc
+    static = return staticDocD
+    dynamic = return dynamicDocD
+
+instance BodySym CSharpCode where
+    type Body CSharpCode = Doc
+    body = liftList bodyDocD
+    bodyStatements = block
+    oneLiner s = bodyStatements [s]
+
+    addComments s = liftA2 (addCommentsDocD s) commentStart
+
+instance BlockSym CSharpCode where
+    type Block CSharpCode = Doc
+    block sts = (lift1List blockDocD endStatement (map (fmap fst) (map state sts)))
+
+instance StateTypeSym CSharpCode where
+    type StateType CSharpCode = Doc
+    bool = return $ boolTypeDocD
+    int = return $ intTypeDocD
+    float = return $ csFloatTypeDoc
+    char = return $ charTypeDocD
+    string = return $ stringTypeDocD
+    infile = return $ csInfileTypeDoc
+    outfile = return $ csOutfileTypeDoc
+    listType p st = liftA2 listTypeDocD st (list p)
+    intListType p = listType p int
+    floatListType p = listType p float
+    boolListType = listType dynamic bool
+    obj t = return $ typeDocD t
+    enumType t = return $ typeDocD t
+    iterator _ = error "Iterator-type variables do not exist in C#"
+
+instance ControlBlockSym CSharpCode where
+    runStrategy l strats rv av = 
+        case Map.lookup l (Map.fromList strats) of Nothing -> error $ "Strategy '" ++ l ++ "': RunStrategy called on non-existent strategy."
+                                                   Just b  -> liftA2 stratDocD b (state resultState)
+        where resultState = case av of Nothing    -> return (empty, Empty)
+                                       Just vari  -> case rv of Nothing  -> error $ "Strategy '" ++ l ++ "': Attempt to assign null return to a Value."
+                                                                Just res -> assign vari res
+
+    listSlice t vnew vold b e s = 
+        let l_temp = "temp"
+            v_temp = var l_temp
+            l_i = "i_temp"
+            v_i = var l_i
+        in
+        (bodyStatements [
+            (listDec l_temp 0 t),
+            for (varDecDef l_i int (getB b)) (v_i ?< getE e) (getS s v_i)
+                (oneLiner $ valState $ v_temp $. (listAppend (vold $. (listAccess v_i)))),
+            (vnew &= v_temp)])
+        where getB Nothing = litInt 0
```

Comment:
you could use `maybe (litInt 0) id b` instead of defining `getB`.

Reply:
Done in bd294b8deb14bcb7196ea30f0fc4dfd819c4be78.